contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,393 @@
+  const cookie = request.cookies.get(ph_cookie_key);
+
+  let distinct_id;
+  if (cookie) {
+    // Use PostHog distinct_id 
+    distinct_id = JSON.parse(cookie.value).distinct_id;
+  } else {
+    // Create new distinct_id
+    distinct_id = crypto.randomUUID();
+  }
+//... rest of code
+```
+
+### Evaluating our redirect test with PostHog
+
+With our distinct ID, we can use the PostHog API to evaluate the `main-redirect` feature flag we set up earlier (because we can’t use PostHog SDKs in Next.js middleware). 
+
+We evaluate the flag by making a POST request to the `https://app.posthog.com/decide?v=3` route (replace `app` with `eu` if you’re on EU Cloud) with your project API key and user distinct ID. From the response, we get the value of the `main-redirect` feature flag and use it to redirect to the right page. Altogether, this looks like this:
+
+```js
+//... rest of code
+
+	const requestOptions = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      api_key: ph_project_api_key,
+      distinct_id: distinct_id
+    })
+  };
+  
+  // Evaluate experiment flag
+  const ph_request = await fetch(
+    'https://app.posthog.com/decide?v=3', // or eu
+    requestOptions
+  );
+  const data = await ph_request.json();
+  const flagResponse = data.featureFlags['main-redirect']
+
+  // Redirect to correct page
+  if (request.nextUrl.pathname=== '/test' && flagResponse === 'control') {
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  if (request.nextUrl.pathname=== '/control' && flagResponse === 'test') {
+    return NextResponse.redirect(new URL('/test', request.url))
+  }
+  return NextResponse.next()
+}
+//... rest of code
+```
+
+### Capturing exposure
+
+To get accurate results for our experiment, we also need to capture a `$feature_flag_called` exposure event after the feature flag has been evaluated, but before redirecting. This requires another POST request like this:
+```js
+//... rest of code, after flag evaluation
+
+// Capture exposure event
+const eventOptions = {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json'
+  },
+  body: JSON.stringify({
+    api_key: ph_project_api_key,
+    distinct_id: distinct_id,
+    properties: {
+      "$feature_flag": 'main-redirect',
+      "$feature_flag_response": flagResponse
+    },
+    event: "$feature_flag_called"
+  })
+};
+
+const eventRequest = await fetch(
+  'https://app.posthog.com/capture',
+  eventOptions
+);
+
+//... rest of code, redirect logic
+```
+
+## Bootstrapping the data
+
+The final piece to our redirect test is bootstrapping the user distinct ID (and flags while we are at it).
+
+> **Why is bootstrapping necessary?** If we didn't bootstrap the distinct ID, PostHog would set a second distinct ID for the same user on the frontend. This disconnects the flag evaluation from the experiment goal tracking on the frontend, breaking our experiment.
```

Comment:
I like the new callout. I still dont know what bootstrapping means, though. Let's clarify that.

Also, this particular sentence is confusing: `This disconnects the flag evaluation from the experiment goal tracking on the frontend, breaking our experiment.`

What does disconnect mean? What does "breaking" mean in this context?

Reply:
added a bootstrap definition and rewrote that paragraph